lib/trino-orc/src/main/java/io/trino/orc/OrcRecordReader.java

```diff
@@ -208,6 +209,15 @@ public OrcRecordReader(
         this.stripes = stripes.build();
         this.stripeFilePositions = stripeFilePositions.build();
 
+        long totalDataLength = 0;
+        for (StripeInfo info : stripeInfos) {
+            StripeInformation stripe = info.getStripe();
+            if (splitContainsStripe(splitOffset, splitLength, stripe)) {
+                totalDataLength += stripe.getDataLength();
```

Comment:
Record this below `totalRowCount` in the above loop

Reply:
Q: What's the scenario when this condition will return false?
```
 if (fileStats.isEmpty() || predicate.matches(numberOfRows, fileStats.get())) {
 ...
 }
```

Reply:
Got it! this is to filter out the files which don't match predicates.